modules/runtime.go

```diff
@@ -0,0 +1,283 @@
+// Copyright 2021 The go-python Authors.  All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package modules
+
+import (
+	"bytes"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"sync"
+
+	"github.com/go-python/gpython/marshal"
+	"github.com/go-python/gpython/py"
+	"github.com/go-python/gpython/vm"
+
+	_ "github.com/go-python/gpython/builtin"
+	_ "github.com/go-python/gpython/math"
+	_ "github.com/go-python/gpython/sys"
+	_ "github.com/go-python/gpython/time"
+)
+
+func init() {
+	// Assign the base-level py.Context creation function while also preventing an import cycle.
+	py.NewContext = NewContext
+}
+
+// context implements py.Context
+type context struct {
+	store     *py.ModuleStore
+	opts      py.ContextOpts
+	closeOnce sync.Once
+	closing   bool
```

Comment:
is `closing` that useful?
it doesn't seem to be really used as of yet.

Reply:
The intent is that Close() can be use reliably and easily, so multiple concurrent Close() support seems important.  And when it comes at no allocation cost, that makes sense to me.